src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * This TaskList class handles the logic of adding and deleting tasks of Duke
+ * 
+ * @param task Task arraylist to hold the current input tasks
+ * 
+ * @author WangYihe
+ * @author E0424695
+ */
+
+package duke.task;
+
+import duke.ui.Ui;
+import duke.exception.DukeException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+//to handle date and time
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+public class TaskList {
+    public final List<Task> task;
+
+    public TaskList() {
+        task = new ArrayList<>();
+    }
+
+    public int getSize() {
+        return task.size();
+    }
+
+    public void add(Task t) {
+        task.add(t);
+    }
+
+    public Task get(int i) {
+        return task.get(i);
+    }
+
+    public boolean isDateFormat(String date, String pattern) {
+        SimpleDateFormat dateFormat = new SimpleDateFormat(pattern);
+        dateFormat.setLenient(false);
+        if (date.length() != 10) {
+            return false;
+        }
+        try {
+            dateFormat.parse(date.trim());
+        } catch (ParseException e) {
+            //TODO: handle exception
+            return false;
+        }
+        return true;
+    }
+
+    public void add(String[] userInput, Ui ui) throws DukeException {
+        switch (userInput[0]) {
+            case "todo":
```

Comment:
According to our coding standard, perhaps there should not be spaces before each "case"? 🤔